cmd/node/node.go

```diff
@@ -358,6 +359,10 @@ func (app *App) Initialize() (err error) {
 		}
 	}
 
+	if app.Config.HARE.N%2 != 0 {
```

Comment:
Whats the reason for enforcing even numbers?

Reply:
so that in other places of the code i can freely use `app.Config.Hare.N/2` without worries.

the idea is if N = 801. what is the majority (the original f+1)? 401? i guess. 402? maybe better.
this is arguably unnecessary. if you feel strongly i will remove.

Reply:
I'm not fully understanding. In what case would you use `app.Config.Hare.N/2` ?

Reply:
What is the number of participants in the devnet?

Reply:
dunno. depends on how widely we 8nvite the community. usually on the order of 100 or less i think.

number of nodes are not the same as the number N here. N is eligibility count, and is proportional to the miner's atx weight. in a network of 50 nodes, like in the systest, we still use 800 for N, and as soon as the eligibility count in the notify round reache s 401, hare terminates.

Reply:
Ok so with eligibility of 800, then the majority would be 401 and  F would be 399, right?

Reply:
> Ok so with eligibility of 800, then the majority would be 401 and F would be 399, right?

the majority would be 401 yes. and F would be 400.
we want termination condition to be 401 so the counts has to reach F+1.

i think i lost the line of questions here :P it started from why do i want N to be even. do you object?

Reply:
I'm just trying to understand how this works.

I think there could be a mistake here though. If you have an eligibility of 800 and your majority is 401 then I think F needs to be 399. Because otherwise if F were 400 and they all abstained from participating then they could permanently halt the protocol, since only 400 would remain and you wouldn't be able to reach the majority of 401.

Reply:
Ok so I think the detail that you were lacking here is that F must be _less_ than half the participants.

I got this from the whitepaper:
>In the following we assume that at most f parties are not honest and at least f + 1 parties are honest. In
particular, f is less than half of all active parties (in each round)

Reply:
> Because otherwise if F were 400 and they all abstained from participating then they could permanently halt the protocol, since only 400 would remain and you wouldn't be able to reach the majority of 401.

hare will fail for this layer if the 400 continue to abstain through iterations until it reached the max iteration limit.
but they will not permanently halt the protocol. the next layer would be already running.

Reply:
For me "majority" is always `N/2 + 1` even if N happens to be odd, but maybe I'm not fully understanding the issue. Is it a problem that `N/2 + 1` and `F` add or do not add up to `N`?

Reply:
Yes that is a problem, the way I've seen this defined before is  `M=N/2 + 1` and `F=N-M` that leaves no room for error. And then it doesn't matter if N is odd or even (if were dealing with ints).

Reply:
Either way `N/2+1` is a majority :D

Reply:
removed. sorry for causing such big confusion.